lib/trino-parquet/src/main/java/io/trino/parquet/reader/ColumnReaderFactory.java

```diff
@@ -104,6 +104,9 @@ public static ColumnReader create(PrimitiveField field, DateTimeZone timeZone, b
                 }
                 throw unsupportedException(type, field);
             }
+            if (type instanceof TimeType && annotation instanceof TimeLogicalTypeAnnotation timeAnnotation && timeAnnotation.getUnit() == MILLIS) {
```

Comment:
I think this should move inside the next if block to get the "primitiveType == INT32" condition as well

Reply:
I was never a fan of huge walls of ifs. And it will probably only get bigger.